package resource.exceptions;

import java.time.Instant;

import javax.servlet.http.HttpServletRequest;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;

import services.exceptions.dataBaseException;
import services.exceptions.resourceNotFoundException;

public class resourceExceptionsHandler {
	@ExceptionHandler(resourceNotFoundException.class)
	public ResponseEntity<standardError> entityNotFound(resourceNotFoundException e, HttpServletRequest request) {
		standardError err = new standardError();
		err.setTimestamp(Instant.now());
		err.setStatus(HttpStatus.NOT_FOUND.value());
		err.setError("Resource not found");
		err.setMessage(e.getMessage());
		err.setPath(request.getRequestURI());
		return ResponseEntity.status(HttpStatus.NOT_FOUND).body(err);
	}
	
	@ExceptionHandler(dataBaseException.class)
	public ResponseEntity<standardError> database(dataBaseException e, HttpServletRequest request) {
		standardError err = new standardError();
		err.setTimestamp(Instant.now());
		err.setStatus(HttpStatus.BAD_REQUEST.value());
		err.setError("Database exception");
		err.setMessage(e.getMessage());
		err.setPath(request.getRequestURI());
		return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(err);
	}
}
